### PR TITLE
storageError port POC

### PR DESCRIPTION
**Pull request by adreed-msft**
Includes a proof of concept implementation for storageError handling.

### CODE REVIEW

File: sdk/storage/azblob/zc_storage_error.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	return err
+}
+
+// defunkifyStorageError is a function that takes the "funky" *runtime.ResponseError and reduces it to a storageError.
+func defunkifyStorageError(responseError *runtime.ResponseError) error {
+	if err, ok := responseError.Unwrap().(*StorageError); ok {
+		// errors.Unwrap(responseError.Unwrap())
+
+		err.response = responseError.RawResponse()
+		err.cause = nil
```
Comment (zezha-msft) [inline sdk/storage/azblob/zc_storage_error.go]: //TODO discuss the cause

File: sdk/storage/azblob/zt_examples_test.go [hunk 2]
```diff
@@ -338,7 +338,7 @@ func ExampleBlobAccessConditions() {
 	// This helper function displays the results of an operation; it is called frequently below.
 	showResult := func(response *DownloadResponse, err error) {
 		if err != nil {
-			if stgErr, ok := err.(StorageError); !ok {
+			if stgErr, ok := err.(*StorageError); !ok {
```
Comment (zezha-msft) [inline sdk/storage/azblob/zt_examples_test.go:376]: Why is it a pointer now?

Comment (adreed-msft) [reply sdk/storage/azblob/zt_examples_test.go:376]: Error() targets a pointer, and the SDK's generated code hands off a pointer to NewResponseError.

As a part of defunkification, we could hand back a non-pointer version.

File: sdk/storage/azblob/zc_storage_error.go (hunk 1 above)
Comment (adreed-msft) [reply sdk/storage/azblob/zc_storage_error.go]: Currently, it's left there more or less as an artifact of old code. It doesn't seem that the new generated code has an underlying cause, so I'm not sure we should really have this anymore.

**Pull request merged**